contracts/Lottery.sol

```diff
@@ -1,12 +1,13 @@
 pragma solidity ^0.5.0;
 pragma experimental ABIEncoderV2;
 
+//import 'openzeppelin-solidity/contracts/math/SafeMath.sol';
+
 contract Lottery {
+    //using SafeMath for uint;
     
     enum UserType {Issuer,Investor}
-    enum LotteryState {NotStarted, Started,Finished}
-    
-    LotteryState public state;
+    enum LotteryState {NotStarted, Started,Finished, Terminated}
```

Comment:
Do we really need a terminated Lottery State here?

I think a NotStarted State should suffice otherwise we will probably need some extra UI handling for this which could have easily been done in the NotStarted state along with the emiited event for Termination.

Unless, I am missing something.

Reply:
There was a method called stopLottery(), which was finishing lottery. My understanding that there will not be stop use-case. It will either will Inprogress or Finished (only after choosing winner). 
However one use-case (probably rare) that owner stops lottery process and in that case, it is premature closure, hence kept this status

Reply:
The system was created in a way to allow registrations and stop them once before picking a winner. If we complete the lottery by picking a winner without effectively closing registrations, we might end up with concurrency problems where you might get a registration request while picking a winner. Would be cleaner to have that rather than simply closing on Winner Selection.